### PR TITLE
Replace Linux binaries with statically linked versions

### PR DESCRIPTION
**Pull request by evanj**
These binaries were compiled with -static, so they do not depend on
any dynamic libraries. These should run without any problems on all
Linux distributions.

The previous binaries were dynamically linked, so they depend on many
libraries. In particular, libpng has been updated, so they do not work
on newer Linux distributions.

### CODE REVIEW

**Comment by evanj** [comment]
I tested this by hacking the URL in `index.js` and running it in Docker against both a current and older versions of Debian. It seems to work. The files are no longer dynamically linked:

```
evanjones@Banff:~/pngquant-bin$ file vendor/linux/x86/pngquant
vendor/linux/x86/pngquant: ELF 32-bit LSB executable, Intel 80386, version 1 (GNU/Linux), statically linked, for GNU/Linux 2.6.32, BuildID[sha1]=12e49648dc4494c7a3a30702d3ae47ce152abd71, stripped
evanjones@Banff:~/pngquant-bin$ file vendor/linux/x64/pngquant 
vendor/linux/x64/pngquant: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), statically linked, for GNU/Linux 3.2.0, BuildID[sha1]=bc62931ca05c5d7b3939437d6b9cd15dbd94cede, stripped
evanjones@Banff:~/pngquant-bin$ ldd vendor/linux/x86/pngquant 
	not a dynamic executable
evanjones@Banff:~/pngquant-bin$ ldd vendor/linux/x64/pngquant 
	not a dynamic executable
```

The disadvantage is that they are now much bigger (~1 MiB instead of ~100 kiB). I could fight with compiler flags a bit more to only statically link some of the libraries, but that will require more work. This seems worth it for the additional reliability:

vendor/linux/x64/pngquant 120648 -> 1412392
vendor/linux/x86/pngquant 100520 -> 1050084

**Comment by evanj** [comment]
The Travis tests have failed but I'm pretty sure it is just flaky errors downloading things. I've retried it by editing the change a couple of times, and one time in failed for Node 8, and this time it failed for Node 4, so I'm pretty sure it is actually working.

**Comment by sindresorhus** [comment]
Why not just update the install commands here: https://github.com/imagemin/pngquant-bin/blob/172fd0edde7c0a5aeead42a94e8761949f780fc0/lib/install.js#L15-L17 ?

**Comment by evanj** [comment]
Correct me if I am wrong, but that only gets used if the built-in binary that is checked in under `vendor/linux/.../pngquant` doesn't work, since this is under the `if (err)` branch on line 7.

This change is to update those checked in binaries with ones that were carefully compiled to be statically linked, so this code should not be needed on most systems. If for some reason it doesn't work, then there is no real need to add `--static`: They are going to need the `libpng` development package installed on their system anyway.

**Comment by gpspake** [comment]
We're seeing errors that pngquant requires GLIBC_2.14 (redhat 6.9). Would this likely resolve that issue?

**Comment by evanj** [comment]
Yes it should! I'd love help testing this though. I *think* you can test it by adding the following to requirement your `package.json`:

```
"pngquant-bin": "git://github.com/evanj/pngquant-bin.git#static-binaries",
```

I *think* you may be able to manually do an "npm install" with that argument to force it to replace the version, but I'm not super familiar with npm, so you may need to remove/install the package to force it to pull that version again?

If it works or does not work for you, let me know!

**Comment by spaivaras** [comment]
Why node community goes such lengths with such cancerous solutions? Why not depend on system level binary from os maintainers? All distros have pngquant which is maintained. Why use some p recompiled version that depends on libraries ~6 years old? Even worse pngquant-bin doesn't even bother to check if system level binary is already available. Anything could be redistributed with compiled binaries. Does node community doesn't care at all that they are getting some random binaries? I understand that evanj tries to fix the problem of dependencies problem for everyone right now, but I TRULY believe that JS should stop redistributing/compiling binaries, but use system level libraries like all other platform does.

File: readme.md [hunk 1]
```diff
@@ -35,6 +35,15 @@ $ pngquant --help
 ```
 
 
+## Updating pre-compiled binaries
+
+The Linux binaries are statically linked so they should work on all Linux distributions. To recompile them:
+
+1. `sudo apt-get install libpng-dev`
+2. `./configure CFLAGS=-static && make && cp pngquant pngquant-64`
+3. Repeat the above commands, but in a 32-bin docker container started with: docker run -ti -v `pwd`:/source i386/debian:9.3 bash
```
Comment (sindresorhus) [inline readme.md]: I'm on a Mac, can you also include the Docker command for getting the correct 64-bit Linux container?

**Pull request merged**